sourced/ml/cmd/repos2bow.py

```diff
@@ -11,21 +11,22 @@
 
 
 @pause
-def repos2bow_template(args, select=HeadFiles, cache_hook=None, save_hook=None):
+def repos2bow_template(args, select=HeadFiles, cache_hook=None, save_hook=None,
+                       full_pipeline=True):
     log = logging.getLogger("repos2bow")
     extractors = create_extractors_from_args(args)
     session_name = "repos2bow-%s" % uuid4()
     root, start_point = create_uast_source(args, session_name, select=select)
     uast_extractor = start_point.link(Moder(args.mode)) \
         .link(Repartitioner.maybe(args.partitions, args.shuffle)) \
         .link(Cacher.maybe(args.persist))
-    if cache_hook is not None:
+    if cache_hook is not None and full_pipeline:
```

Comment:
This will go away then: the cache hook is `None` by default in `index_bag_features`